selfdrive/car/mazda/interface.py

```diff
@@ -44,6 +47,52 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), car_fw=None, expe
       ret.wheelbase = 2.83
       ret.steerRatio = 15.5
 
+    if Params().get_bool('dp_mazda_ti'):
+      if candidate in (CAR.CX5, CAR.CX5_2022):
+        ret.lateralTuning.pid.kiBP = [5.0, 25.0]
+        ret.lateralTuning.pid.kpBP = [5.0, 25.0]
+        ret.lateralTuning.pid.kpV = [0.25,0.28]
+        ret.lateralTuning.pid.kiV = [0.01,0.025]
+        ret.lateralTuning.pid.kf = 0.00008
+
+        ret.lateralTuning.init('indi')
+        ret.lateralTuning.indi.innerLoopGainBP = [5.0, 35]
+        ret.lateralTuning.indi.innerLoopGainV = [4.5, 6.0]
+        ret.lateralTuning.indi.outerLoopGainBP = [5, 35]
+        ret.lateralTuning.indi.outerLoopGainV = [3.0, 6]
+        ret.lateralTuning.indi.timeConstantBP = [2, 35]
+        ret.lateralTuning.indi.timeConstantV = [0.2, 1.5]
+        ret.lateralTuning.indi.actuatorEffectivenessBP = [0, 25]
+        ret.lateralTuning.indi.actuatorEffectivenessV = [2.0, 1]
+      elif candidate in [CAR.CX9, CAR.CX9_2021]:
+        ret.lateralTuning.pid.kiBP = [8.0, 30.0]
+        ret.lateralTuning.pid.kpBP = [8.0, 30.0]
+        ret.lateralTuning.pid.kpV = [0.10,0.22]
+        ret.lateralTuning.pid.kiV = [0.01,0.019]
+        ret.lateralTuning.pid.kf = 0.00006
+      elif candidate == CAR.MAZDA3:
+        ret.lateralTuning.pid.kiBP = [5.0, 25.0]
+        ret.lateralTuning.pid.kpBP = [5.0, 25.0]
+        ret.lateralTuning.pid.kpV = [0.25,0.28]
+        ret.lateralTuning.pid.kiV = [0.01,0.025]
+        ret.lateralTuning.pid.kf = 0.00008
```

Comment:
no need to call for it twice. 

you can 
 ```if  in candidate in [CAR.CX9, CAR.CX9_2021]:
        ret.lateralTuning.pid.kiBP = [5.0, 25.0]
        ret.lateralTuning.pid.kpBP = [5.0, 25.0]
        ret.lateralTuning.pid.kpV = [0.25,0.28]
        ret.lateralTuning.pid.kiV = [0.01,0.025]
        ret.lateralTuning.pid.kf = 0.00008```

Reply:
Oh wait, we don't need these pid tuning values here, we can just use the new torque params which work fine. I guess I grabbed an old file by accident, didn't even notice. Will fix soon.

Reply:
Updated in last commit, should be right now.. probably.. could also be totally screwed up somewhere and I'm not seeing it. Anyway, this is how it's set up for us on the branch we use for TI users and it's been working just fine. @rav4kumar Could you tell me if I missed anything here? I know this is slow going but I've not had a whole lot of free time. 😅

Reply:
Oh I forgot, dp let's you switch controllers.. I guess we do need the pid tuning values here after all. Will add them back later today I guess, I'll clean it up a bit too.

Reply:
speaking of which, which is the best controller for TI?
I'm working on an alternative controller feature which allows user to switch to a different controller at a different set speed.

Reply:
We have been using the new torque controller since 0.8.15, but I didn't really notice a big difference, maybe a little smoother, but it would be hard to say if one is better than the other.

Reply:
How does it handle when making sharp turns at low speeds?

Reply:
Come to think of it, it seems like low speed was a little jerky with the TI.. I would have to go back and test, it's been a little while. Also I very rarely use it at low speed since I don't have stop and go, but that would be different with dragonpilot.